### ISSUE TITLE
Wagtail Admin: Help Text Overlaps StreamField Within Streamfield

### ISSUE THREAD

**Issue by FlipperPA** [wagtail/wagtail#4247]
### Issue Summary

In the admin, the help text for a parent StreamField overlaps the nested StreamField. A picture is worth a thousand words - see under "Headline":

![image](https://user-images.githubusercontent.com/68164/35934487-c9186e9a-0c0b-11e8-852c-6d905ea629d7.png)

### Steps to Reproduce

1. Create a Wagtail Page with a `BodyBlock`. Here's the `blocks.py` file we use with `BodyBlock` and its dependencies:

```python
from wagtail.contrib.table_block.blocks import TableBlock
from wagtail.core.blocks import (
    StructBlock,
    TextBlock,
    StreamBlock,
    RichTextBlock,
    PageChooserBlock,
    URLBlock,
    BooleanBlock,
    ListBlock,
)
from wagtail.embeds.blocks import EmbedBlock
from wagtail.images.blocks import ImageChooserBlock
from wagtailcodeblock.blocks import CodeBlock


class LinkBlock(StructBlock):
    title = TextBlock(required=False)
    related_page = PageChooserBlock(required=False)
    external_url = URLBlock(required=False)
    open_in_new_tab = BooleanBlock(required=False)
    caption = TextBlock(required=False)


class CaptionedImageBlock(StructBlock):
    image = ImageChooserBlock()
    caption = TextBlock(required=False)

    class Meta:
        icon = 'fa-image'
        template = 'standard/blocks/captioned_image.html'
        help_text = 'Select an image and add a caption (optional).'


class ContentStreamBlock(StreamBlock):
    paragraph = RichTextBlock(
        icon='fa-paragraph',
        features=['bold', 'italic', 'ol', 'ul', 'link', 'document-link', 'code', ],
    )
    heading = TextBlock(icon='fa-header', template='standard/blocks/heading.html')
    image = CaptionedImageBlock()
    table = TableBlock(icon='fa-table')
    embed = EmbedBlock(icon='fa-youtube-play')
    code = CodeBlock(label='Code snippet')
    link_list = ListBlock(
        LinkBlock(label='Link'),
        label='List Of Links',
        template='standard/blocks/link_list.html',
        help_text='Create a list of links.')

    class Meta:
        template = 'standard/blocks/streamfield.html'


class SectionBlock(StructBlock):
    headline = TextBlock(help_text='Write a title for this section.')
    body = ContentStreamBlock(help_text='The section content goes here.')

    class Meta:
        icon = 'emoji-bookmark-tabs'
        template = 'standard/blocks/section.html'
        help_text = 'Sections divide the page into digestible parts.'


class BodyBlock(StreamBlock):
    section = SectionBlock()

    class Meta:
        template = 'standard/blocks/streamfield.html'
        help_text = 'The main page body.'
```

### Technical details

* Python version: 3.6.4
* Django version: 1.11.10
* Wagtail version: 2.0b1
* Browser version: Latest Chrome & Firefox (screenshot from Firefox, same behavior in Chrome)

Thank you, as always.

**Comment by gasman** [comment]
Thanks for the report @FlipperPA. Confirmed here - looks like this was introduced in https://github.com/wagtail/wagtail/commit/a03521f575265168da240fb5f7da106fcab2e755.

The previous rendering wasn't particularly pretty either, and I suspect the original CSS wasn't written with this configuration of elements in mind...
<img width="894" alt="screen shot 2018-02-13 at 00 52 10" src="https://user-images.githubusercontent.com/85097/36128065-41a0cfd8-1058-11e8-88e8-fc13b921b51c.png">

The help text overlapping the field certainly does seem like a regression, though. @thibaudcolas, how feasible is it to fix this for 2.0? (I realise the StreamField CSS is really brittle, and the rewrite is still in progress, so if we do have to let this one slide, so be it...)

**Comment by FlipperPA** [comment]
Certainly not the end of the world, and I realize the challenge of CSS with nested StreamField configurations is challenging, to say the least. Maybe switching on-hover tether tooltips to an existing element?

**Comment by thibaudcolas** [comment]
I'll have a look at this now. I had to refactor this as part of making the editor toolbar `position: sticky;`, hopefully it's just this one scenario I overlooked while rewriting the styles and it'll be an easy fix.

**Comment by thibaudcolas** [comment]
Should be addressed in #4281.

**Comment by gasman** [comment]
Fixed in #4281
